android/sdl_android/src/main/java/com/smartdevicelink/proxy/SdlProxyBase.java

```diff
@@ -2103,7 +2116,102 @@ private void handleErrorsFromInternalMessageDispatcher(String info, Exception e)
 		_proxyListener.onError("Proxy callback dispatcher is down. Proxy instance is invalid.", e);
 	}
 	/************* END Functions used by the Message Dispatching Queues ****************/
-	
+
+
+	private OnRPCNotificationListener mEncryptionReqListener = new OnRPCNotificationListener() {
+		@Override
+		public void onNotified(RPCNotification notification) {
+			List<PermissionItem> permissionItems = ((OnPermissionsChange)notification).getPermissionItem();
+			Set<String> encryptedRpcs = new HashSet<>();
+			Boolean requireEncryptionAppLevel = ((OnPermissionsChange) notification).getRequireEncryption();
+			if (permissionItems != null && !permissionItems.isEmpty()) {
+				for (PermissionItem permissionItem : permissionItems) {
+					if (permissionItem != null) {
+						if (Boolean.TRUE.equals(permissionItem.getRequireEncryption())) {
+							if (requireEncryptionAppLevel == null || requireEncryptionAppLevel == true) {

```

Comment:
Can we do the `requireEncryptionAppLevel` check before the for loop. That way don't have to loop if not necessary

Reply:
This logic does not applied anymore since encryption can be necessary if requireEncryptionAppLevel is missing (null).

Reply:
Right but cannot we check both conditions `if (requireEncryptionAppLevel == null || requireEncryptionAppLevel == true)` before the loop?